apps/core/lib/core/snitch/address.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Core.Snitch.Address do
+  @moduledoc """
+  Models an Address
+
+  An `Address` must contain a reference to `Core.Snitch.Country` and only if the
+  country has states a reference to a `Core.Snitch.State`. This means some
+  Addresses might not have a State.
+  """
+
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  @type t :: %__MODULE__{}
+
+  schema "snitch_addresses" do
+    # Do we really need first and last name separately in Address??
+    field(:first_name, :string)
+    field(:last_name, :string)
```

Comment:
Who does these first_name, last_name referred to

Reply:
In spree, the user's name is simply copied in here. It's an editable field in the UI.

Reply:
Address is always stored in modular form, so they make total sense @oyeb